### PR TITLE
AzureMonitor: Rename labels for query type dropdown

### LINKED ISSUE #29847: AzureMonitor: rename sections to Metrics, Logs and App Insights
The current dropdown of options in the Azure Monitor data source has four entries:

image

It is quite confusing for users and doesn't map well to the metaphors of the Grafana UI.

The proposal is to:
- Azure Monitor -> Metrics
- Azure Log Analytics -> Logs
- Application Insights -> Application Insights

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Renames the Service/queryType dropdown options according to #29847 

Azure Monitor --> Metrics
Azure Log Analytics--> Logs
Application Insights -> _no change_
Insights Analytics -> _no change_

image

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/29847

### CODE REVIEW

**Review by kylebrandt — COMMENTED** [review]
Will need documentation updated at https://github.com/grafana/grafana/blob/master/docs/sources/datasources/azuremonitor.md

**Comment by kylebrandt** [comment]
Also rename should probably end up in change log, so when ready and setting a milestone, also do add to changelog.

File: docs/sources/datasources/azuremonitor.md [hunk 1]
```diff
@@ -67,20 +67,22 @@ az ad sp create-for-rbac -n "http://localhost:3000"
...
+- `Logs`
 - `Insights Analytics`
 
-The query editor changes depending on which one you pick. Azure Monitor is the default.
+The query editor changes depending on which one you pick. Metrics is the default.
 
 Starting in Grafana 7.1, Insights Analytics replaced the former edit mode from within Application Insights.
 
-## Query the Azure Monitor service
+Starting in Grafana 7.4, the Azure Monitor query type was renamed to Metrics and Azure Logs Analytics was renamed to Logs.
```
Comment (osg-grafana) [inline docs/sources/datasources/azuremonitor.md:81]: @oddlittlebird Line 81 is an example of something that I would expect to be in release notes.

**Review by osg-grafana — APPROVED** [review]

**Comment by kylebrandt** [comment]
@joshhunt Also analytics reference in https://github.com/grafana/grafana/blob/master/docs/sources/getting-started/timeseries-dimensions.md#multiple-dimensions in the first "Note". (Also if you could please remove the SQL planned to be added for 7.2 and replace with planned in a "future version", (or just remove the SQL parts of that note together), or change to version 8 (Whatever docs folks thinks is best, I listed in order of what I think might be best).

**Comment by kylebrandt** [comment]
I added some labels that might need to be renamed after merging.

**Comment by kylebrandt** [comment]
Also maybe this is just part of the plan, or perhaps my info is out of date, but I thought it was:

Azure Monitor --> Metrics
Azure Log Analytics--> Logs
Insights Analytics--> Application Insights (Logs)
Application Insights-->  Application Insights (Metrics)

Which might make it all make a little more sense, since the two Logs services uses KQL (Kusto).

**Comment by joshhunt** [comment]
@daniellee would love to get your view on the labels @kylebrandt mentioned. They make sense and I'm indifferent either way as I changed the component to seperate the labels in the UI from the actual value stored

File: docs/sources/datasources/azuremonitor.md (hunk 1 above)
Comment (oddlittlebird) [reply docs/sources/datasources/azuremonitor.md:81]: Yes, I agree. @joshhunt, can you help us write up a short blurb for this? It does not need to be long.

**Comment by daniellee** [comment]
@kylebrandt slight modification to the plan.

So this is totally correct:

Azure Monitor --> Metrics
Azure Log Analytics--> Logs

This is going to happen by us migrating Insights analytics to Logs (and not renaming):

Insights Analytics--> Application Insights (Logs)

And App Insights stays the same for now. So we will have Metrics, Logs and Application Insights after these two changes (the rename and the migration). Application Insights will eventually go under Metrics but not sure how do to that yet.

File: docs/sources/datasources/azuremonitor.md (hunk 1 above)
Comment (joshhunt) [reply docs/sources/datasources/azuremonitor.md:81]: @oddlittlebird Sure! What about:

> Azure Monitor query type was renamed to Metrics and Azure Logs Analytics was renamed to Logs to match the service names in Azure and align the concepts with the rest of Grafana.

I'll let you wordsmith if need be 😅

Comment (oddlittlebird) [reply docs/sources/datasources/azuremonitor.md:81]: Perfect, thank you! I have added it to the What's New in 7.4? Gdoc.

**Review by kylebrandt — APPROVED** [review]

**Comment by joshhunt** [comment]
(oops sorry, wrong pr)

**Pull request merged**